ConditionalTranslation.md

```diff
@@ -0,0 +1,297 @@
+  Its grammar is a subset of normal WGSL [expressions](https://www.w3.org/TR/WGSL/#expressions). it must be one of:
+  * a *translate-time feature*,
+  * a [logical expression](https://www.w3.org/TR/WGSL/#logical-expr): logical not (`!`), short-circuiting AND (`&&`), short-circuiting OR (`||`),
+  * a [parenthesized expression](https://www.w3.org/TR/WGSL/#parenthesized-expressions),
+  * a boolean literal value (`true` or  `false`).
+
+* **Translate-time scope**: The *translate-time scope* this is an independent scope from the [*module scope*](https://www.w3.org/TR/WGSL/#module-scope), meaning it cannot see any declarations from the source code, and its identifers are independent.
+
+* **Translate-time feature**: A *translate-time feature* is an identifier that evaluates to a boolean. It is set to `true` if the feature is *enabled* during the translation phase and `false` if the feature is *disabled*.
+
+* **Translate-time attribute**: A *translate-time attribute* is parametrized by a *translate-time expression*. It is eliminated after the translation but can affect the syntax node it decorates.
+
+## Location of *Translate-time attributes*
+
+A *translate-time attribute* can appear in all places where the WGSL grammar allows attributes, except if removal of the decorated syntax node would lead to syntactically incorrect code.
+The grammar is extended to allow attributes in several locations previously not allowed by the WGSL grammar. These extensions are specified in section [updated grammar](#updated-grammar).
+
+### Summary
+
+1. The grammar is extended to allow *translate-time attributes* before the following syntax nodes:
+   * const value declarations
+   * variable declarations
+   * directives
+   * switch clauses
+   * else if clauses and else clauses
+   * all statements that did not allow attributes:
+     * assignment statements
+     * increment and decrement statements
+     * break statements
+     * break-if statements
+     * continue statements
+     * continuing statements
+     * return statements
+     * discard statements
+     * function call statements
+     * const assertion statements
+
+2. A *translate-time attribute* CAN decorate the following syntax nodes:
+   * structure declarations
+   * structure members
+   * function declarations
+   * function formal parameter declarations
+   * variable declarations and value declarations
+   * all statements, except those disallowed
+   * directives
+
+3. A *translate-time attribute* CANNOT decorate the following syntax nodes, even if the WGSL grammar allows attributes before these syntax nodes:
+   * function return types
+   * the body (part surrounded by curly braces) of:
+     * function declarations
+     * switch statements
+     * switch clauses
+     * loop statements
+     * for statements
+     * while statements
+     * if/else statements
```

Comment:
confusing that you can condition `else if` but not `if`..

Reply:
you can condition if, but it's already in the grammar. That's why it's not mentionned here.

Overall seeing your comments I need to rephrase that this section refers to edits to the grammar. I'll add a more human-friendly explanation too.